lib/elixir/lib/option_parser.ex

```diff
@@ -327,30 +327,56 @@ defmodule OptionParser do
     {:error, argv}
   end
 
-  defp next(["-" <> option | rest] = argv, aliases, switches, strict) do
+  # Handles --foo or --foo=bar
+  defp next(["--" <> option | rest], _aliases, switches, strict) do
     {option, value} = split_option(option)
-    original = "-" <> option
-    tagged = tag_option(option, switches, aliases)
+    tagged = tag_option(option, switches)
+    do_next(tagged, value, "--" <> option, rest, switches, strict)
+  end
+
+  # Handles -a, -abc, -abc=something
+  defp next(["-" <> option = original | rest] = argv, aliases, switches, strict) do
+    {option, value} = split_option(option)
+    original_option = "-" <> option
 
     cond do
       negative_number?(original) ->
         {:error, argv}
-      strict and not option_defined?(tagged, switches) ->
-        {:undefined, original, value, rest}
-      true ->
-        {option, kinds, value} = normalize_option(tagged, value, switches)
-        {value, kinds, rest} = normalize_value(value, kinds, rest, strict)
-        case validate_option(value, kinds) do
-          {:ok, new_value} -> {:ok, option, new_value, rest}
-          :invalid         -> {:invalid, original, value, rest}
+      String.contains?(option, ["-", "_"]) ->
+        {:undefined, original_option, value, rest}
+      String.length(option) > 1 ->
+        if (opt = get_option(option)) && (alias = aliases[opt]) do
+          IO.warn "multi-letter aliases are deprecated, got: #{inspect(opt)}"
+          do_next({:default, alias}, value, original_option, rest, switches, strict)
+        else
+          {first, other} = String.next_codepoint(option)
+          tagged = tag_single_letter_alias(first, aliases)
```

Comment:
Not sure I'm following. We're calling `do_next/6` so we won't fallback to the clause below here. However, we could probably call `next/4` recursively here, is that what you meant? :)